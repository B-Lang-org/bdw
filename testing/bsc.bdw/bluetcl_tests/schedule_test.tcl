namespace import ::Bluetcl::*

# Sets the specified flag
puts [flags set {-verilog}]
puts {----------}

# Loads the specified module
puts [module load mkS]

# Returns the urgency dependencies 
puts [schedule urgency mkS]

# Returns the rule/method execution order
puts [schedule execution mkS]

# Returns 
puts [schedule methodinfo mkS]

# Returns
puts [schedule pathinfo mkS]

# Returns warnings accured during scheduling for the loaded module
puts [schedule warnings mkS]

puts {----------}

# Loads the specified module
puts [module load mkM]

# Returns the urgency dependencies 
puts [schedule urgency mkM]

# Returns the rule/method execution order
puts [schedule execution mkM]

# Returns 
puts [schedule methodinfo mkM]

# Returns
puts [schedule pathinfo mkM]

# Returns warnings accured during scheduling for the loaded module
puts [schedule warnings mkM]

