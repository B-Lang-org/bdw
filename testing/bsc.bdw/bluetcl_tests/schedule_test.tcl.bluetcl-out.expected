-verilog
----------
mkS
{wset {}} {wget {}} {RL_c1 {}} {RL_c2 RL_c1}
wset wget RL_c1 RL_c2
{wset {{wset C} {wget <R}}} {wget {{wset >R} {wget CF}}}
{{wset_datain EN_wset} wget}
{{Test.bsv 113 8} G0010 {Warning: "Test.bsv", line 113, column 8: (G0010)
  Rule "c1" was treated as more urgent than "c2". Conflicts:
    "c1" cannot fire before "c2": calls to rg1.write vs. rg1.read
    "c2" cannot fire before "c1": calls to rg1.write vs. rg1.read
}} {{Test.bsv 123 9} G0021 {Warning: "Test.bsv", line 123, column 9: (G0021)
  According to the generated schedule, rule `c2' can never fire.
}}
----------
mkM
{_write {}} {_read {}} {RL_rHello {}} {RL_rWorld {}}
_read RL_rHello _write RL_rWorld
{_write {{_write <R} {_read >}}} {_read {{_write <} {_read CF}}}


